File: Java/[2 sem] 2DWorldSimulator/code/Wolf.java
public class Wolf extends Animal {
    private static final int WOLF_STRENGTH = 9;
    private static final int WOLF_INITIATIVE = 5;
    private static final char WOLF_SYMBOL = 'W';

    public Wolf(int x, int y, World world) {
        super(x, y, world, WOLF_STRENGTH, WOLF_INITIATIVE, WOLF_SYMBOL, 0);
    }

    @Override
    public boolean isWolf() {
        return true;
    }

    @Override
    public String getSpecieName() {
        return "Wolf";
    }
}

File: C(++)/[2 sem] 2DWorldSimulator/code/src/worldObjects/animals/Wolf.cpp
//
// Created by szymo on 4/16/2023.
//

#include ".headers/Wolf.h"

#define WOLF_STRENGTH 9
#define WOLF_INITIATIVE 5
#define WOLF_SYMBOL 'W'


Wolf::Wolf(int x, int y, World *world) : Animal(x, y, world, WOLF_STRENGTH, WOLF_INITIATIVE, WOLF_SYMBOL)
{}

bool Wolf::isWolf() {
    return true;
}

std::string Wolf::getSpecieName() const {
    return "Wolf";
}